modmail/utils/cogs.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass
+from enum import IntEnum
+
+
+class BotModes(IntEnum):
+    """
+    Valid modes for the bot.
+
+    These values affect logging levels, which logs are loaded, and so forth.
+    """
+
+    production = int("1", 2)
+    develop = int("10", 2)
+    plugin_dev = int("100", 2)
+
+
+BOT_MODES = BotModes
+
+
+@dataclass()
+class ExtMetadata:
+    """Ext metadata class to determine if extension should load at runtime depending on bot configuration."""
+
+    # prod mode
+    # set this to true if the cog should always load
+    production: bool = False
+    # load if bot is in development mode
+    # development mode is when the bot has its metacogs loaded, like the eval and extension cogs
+    develop: bool = False
+    # plugin development mode
+    # used for loading bot plugins that help with plugin debugging
+    plugin_dev: bool = False
+
+
+def calc_mode(metadata: ExtMetadata) -> int:
+    """Calculate the combination of different variables and return the binary combination."""
+    mode = int(getattr(metadata, "production", False))
+    mode = mode + int(getattr(metadata, "develop", False) << 1) or 0
```

Comment:
+=

Reply:
Also no need for getattr here unless we want to meta-code this. ( for name in ...) which we shouldn't. We should be directly translating using the enum members values and names. We can define `def _generate_next_value_(name, start, count, last_values):` to autogenerate these and just provide names, and `__members__` to determine what possible values there are... in order, and then we don't have to hardcode any of this

Reply:
See comment above this set with BitwiseAutoEnum, you can list() an enum to get it's members so use that and .name/.value to do this magically